### ISSUE TITLE
After Search, Link to Reviews Malformed

### ISSUE THREAD

**Issue by bhsmither** [cubecart/v6#1006]
See:
https://forums.cubecart.com/topic/50871-product-reviews-not-working-right/

In Catalogue class, line 1343, when `$link` is being constructed, to populate a search results page (that is, `$product_view` is false), the currentPage has the `search[]` key still present, to which is improperly appended some more querystring.

Suggest :

```
$link = currentPage(array('cat_id'), array('_a' => 'product', 'product_id' => $product['product_id']), false).'#reviews';
```

The '_a' key has been removed from `$excluded` because its value will be replaced and needs to not be excluded.

The `search[]` key also remains, but is ignored everywhere except the breadcrumb.

The resulting `$link` will not match SEOable() -- perhaps not an issue.

**Comment by bhsmither** [comment]
Perhaps it is an issue because then CANONICAL acquires a wrong URL.

**Comment by abrookbanks** [comment]
Pretty sure this relates to #821

**Comment by briansandall** [comment]
Is this still an issue? Current code for building the review info link starts on line 1432:
```
if (!$product_view) {
	$link = $GLOBALS['seo']->buildURL('prod', $product['product_id'], '&') . '#reviews';
} else {
	$link = '#reviews';
}
```

**Comment by bhsmither** [comment]
Probably fixed with this: https://github.com/cubecart/v6/issues/1391

**Comment by briansandall** [comment]
Yes, that was my intent as I couldn't replicate the problem. Trying to clear out some of these issues :)

**Comment by abrookbanks** [comment]
Fixed with #1391
